janusgraph-core/src/main/java/org/janusgraph/core/ConfiguredGraphFactory.java

```diff
@@ -243,7 +244,9 @@ private static void removeGraphFromCache(final String graphName) {
     private static void removeGraphFromCache(final JanusGraph graph) {
         final JanusGraphManager jgm = JanusGraphManagerUtility.getInstance();
         Preconditions.checkNotNull(jgm, JANUS_GRAPH_MANAGER_EXPECTED_STATE_MSG);
-        jgm.removeGraph(((StandardJanusGraph) graph).getGraphName());
+        final String graphName = ((StandardJanusGraph) graph).getGraphName();
+        jgm.removeGraph(graphName);
+        jgm.removeTraversalSource(graphName + "_traversal");
```

Comment:
Same here, if we would have a dedicated method to convert `graphName` to `traversalSourceName` we could change this line to:
```
jgm.removeTraversalSource(toTraversalSourceName(graphName));
```